SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -597,6 +614,9 @@ - (void)handleProtocolRegisterSecondaryTransportNAKMessage:(SDLProtocolMessage *
     SDLLogW(@"Received Register Secondary Transport NAK frame");
 
     dispatch_async(self.stateMachineQueue, ^{
+        if ([self.stateMachine.currentState isEqualToEnum:SDLSecondaryTransportStateRegistered]) {
+            [self sdl_handleTransportUpdateWithPrimaryAvailable:YES secondaryAvailable:NO];
```

Comment:
It is possible to enter state "Reconnecting" while the transport is still open and when the transport is closed. If the transport is closed, then we would not want to call `sdl_handleTransportUpdateWithPrimaryAvailable:secondaryAvailable` because the transport can't be updated due to it being closed. When we get a `Register Secondary Transport NAK`, the TCP transport is still open so we can try to send another  register secondary transport request.